internal/cache/badger/badger.go

```diff
@@ -177,38 +155,29 @@ func (c *Cache) Get(keys []string) ([]bool, error) {
 }
 
 // Put adds or updates elements inside the cache with the specified TTL and returns an error if the operation failed
-func (c *Cache) Put(keys []string, ttl time.Duration) error {
-	itemsByHashRange, err := c.hasher.GetKeysByHashRange(keys)
-	if err != nil {
-		return fmt.Errorf("cache put keys: %w", err)
-	}
-
+func (c *Cache) Put(keys []string, ttl time.Duration, hashRange uint32) error {
 	if c.batchWriterEnabled {
 		bw := c.cache.NewWriteBatch()
-		for hashRange, keys := range itemsByHashRange {
-			for _, key := range keys {
-				entry := badger.NewEntry(c.getKey(key, hashRange), nil)
-				if ttl > 0 {
-					entry = entry.WithTTL(c.getTTL(ttl))
-				}
-				if err := bw.SetEntry(entry); err != nil {
-					return fmt.Errorf("failed to put key %s: %w", key, err)
-				}
+		for _, key := range keys {
+			entry := badger.NewEntry(c.getKey(key, hashRange), nil)
+			if ttl > 0 {
+				entry = entry.WithTTL(c.getTTL(ttl))
```

Comment:
I'm thinking that it might make sense to use one TTL per batch (as in one per Put call).
We still won't have big blocks of data expiring at the same time but at least if a TTL triggers it can delete at least a small batch. It seems also less wasteful to run rand on every single key but rather on a batch.